stake-pool/cli/src/main.rs

```diff
@@ -1611,6 +1856,61 @@ fn main() {
                           Defaults to the token-owner's associated pool token account. \
                           Creates the account if it does not exist."),
             )
+            .arg(
+                Arg::with_name("referrer")
+                    .validator(is_pubkey)
+                    .value_name("ADDRESS")
+                    .takes_value(true)
+                    .help("Account to receive the referral fees for deposits. \
+                          Defaults to the pool manager."),
+            )
+        )
+        .subcommand(SubCommand::with_name("deposit-sol")
+            .about("Deposit SOL into the stake pool in exchange for pool tokens")
+            .arg(
+                Arg::with_name("pool")
+                    .index(1)
+                    .validator(is_pubkey)
+                    .value_name("POOL_ADDRESS")
+                    .takes_value(true)
+                    .required(true)
+                    .help("Stake pool address"),
+            ).arg(
+                Arg::with_name("amount")
+                    .index(2)
+                    .validator(is_amount)
+                    .value_name("AMOUNT")
+                    .takes_value(true)
+                    .help("Amount in SOL to deposit into the stake pool reserve account."),
+            )
+            .arg(
+                Arg::with_name("from")
+                    .long("from")
+                    .validator(is_keypair)
```

Comment:
If people want to use this with ledger, we'll need to support `is_keypair_or_ask_keyword` -- what do you think?

Reply:
This doesn't really grant support for ledger, does it?

That really requires remote wallet

Reply:
Anw, I've added the ask keyword in case ppl want it

Reply:
We're using remote wallet already, right?

Reply:
Then isn't what you want, is_signer?

Reply:
Oh, I see, you always ask the config for this.